zypp/target/rpm/RpmDb.cc

```diff
@@ -331,6 +334,14 @@ void RpmDb::initDatabase( Pathname root_r, Pathname dbPath_r, bool doRebuild_r )
     ZYPP_THROW(RpmInvalidRootException(root_r, dbPath_r));
   }
 
+  if ( dbPath_r == "/var/lib/rpm"
+    && ! PathInfo( root_r/"/var/lib/rpm" ).isExist()
+    && PathInfo( root_r/"/usr/lib/sysimage/rpm" ).isDir() )
+  {
+    WAR << "Rpm package was deleted? Injecting missing rpmdb compat symlink." << endl;
+    filesystem::symlink( "../../usr/lib/sysimage/rpm", root_r/"/var/lib/rpm" );
```

Comment:
Why do you use a relative path here for the symlink and not the one you checked for in line 339?

Reply:
Also we could use root_r / dbPath_r instead of the hard coded string

Reply:
Symlinks must be relative otherwise they do not work in rooted environments. The ocation must be valid no matter if or where the system is rooted to.

Reply:
Makes sense, thanks for clarifying